tempesta_fw/t/functional/helpers/control.py

```diff
@@ -258,13 +259,17 @@ def start(self):
         cmd = '%s/scripts/tempesta.sh --start' % self.workdir
         env = { 'TFW_CFG_PATH': self.config_name }
         self.node.run_cmd(cmd, timeout=30, env=env, err_msg=(self.err_msg % 'start'))
+        self.state = "started"
```

Comment:
I have tried such approach, it also has an issue. Imagine, you've interrupted (repeated `^C`) a test just in the middle. Then TempestaFW and Nginx will remain alive. If you try to start a new test it will be failed on `setUp()` due to errors on starting TempestaFW and Nginx. Without saving the state you can simply run the same test once again to cleanup the environment.

The problem @intelfx highlighted - exceptions in `tearDown()`, which makes clenup procedure  interrupted if  one of it's steps failed. Every `node.run_cmd()` call may throw an exception and storing the state actually doesn't solve the problem. I even think, that it doesn't provide any advantages.